### PR TITLE
Use case onboarding wizard

### PR DESCRIPTION
**Pull request by neallred**
#### Summary
Add use case onboarding for first admin. Added a UseCaseOnboarding feature flag. When off, no functionality should change for first admin (and other users functionality should be unchanged regardless).

Deferred QA because we are going to test this with other PRs post merge.

Not implemented:
We decided to implement the account creation screen (used in the self-hosted first admin case) in a separate PR because that code is somewhat complicated (tied to a general header/footer template component, also would need changes for multi factor authentication options, and has duplicated code between itself and the login component) and would further increase the scope of work.

#### Ticket Link
https://community-daily.mattermost.com/boards/workspace/fpbb7orqpiyfxqgg6fcu91y5xo/b6ropkdiowj8eugbt8kifrew79h/vptcho591nfry8rgfygxe8f7tbr/cyg1wqa6j97gat86c6ursd7hcqa

#### Related Pull Requests
Server PR: https://github.com/mattermost/mattermost-server/pull/19446

#### Screenshots
Self hosted:

TBD

Cloud:

TBD

#### Release Note

```release-note
Add Use Case Onboarding wizard for first admin.
```

### CODE REVIEW

**Comment by hanzei** [comment]
cc @DHaussermann

File: utils/channel_utils.tsx [hunk 1]
```diff
@@ -98,3 +109,23 @@ export function joinPrivateChannelPrompt(team, channel, handleOnCancel = true) {
         return result;
     };
 }
+
+export function makeNewEmptyChannel(displayName: string, teamId: string): Channel {
+    return {
+        team_id: teamId,
+        name: displayName.split(' ').join('-').toLowerCase(),
```
Comment (JulienTant) [inline utils/channel_utils.tsx:116]: Do we have a tool/function that does some normalization? Just wondering if there's a way that some non-desired characters end up here (eg cyrilic or greek letters)

File: components/root/root.jsx [hunk 2]
```diff
@@ -264,11 +268,34 @@ export default class Root extends React.PureComponent {
...
+            return;
+        }
+
+        GlobalActions.redirectUserToDefaultTeam();
+    }
+
     componentDidMount() {
         this.mounted = true;
         this.props.actions.loadMeAndConfig().then((response) => {
             if (this.props.location.pathname === '/' && response[2] && response[2].data) {
```
Comment (JulienTant) [inline components/root/root.jsx:297]: could you explicit `response[2]`?

File: components/preparing_workspace/use_case.tsx [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+                                        icon: <PlaybooksSVG/>,
+                                        id: t('onboarding_wizard.use_case.playbooks'),
+                                        defaultMessage: 'Processes, workflows, and automation',
+                                        checked: props.options.playbooks,
+                                    },
+                                ]}
+                            />
+                        </PageBody>
+                        <div>
+                            <button
```
Comment (JulienTant) [inline components/preparing_workspace/use_case.tsx:115]: I'm wondering if there's a way to extract the buttons to make it a reusable component over the pages, but not sure it's worth it

**Review by JulienTant — CHANGES_REQUESTED** [review]
Impressive work, good job @neallred! I did mark this PR as "Request changes" mostly to make sure I get back to it after you react to my questions and remarks.

File: utils/constants.jsx [hunk 4]
```diff
@@ -128,6 +128,12 @@ export const Preferences = {
 
     // A/B test preference value
     AB_TEST_PREFERENCE_VALUE: 'ab_test_preference_value',
+
+    ONBOARDING: 'onboarding',
+};
+
+export const OnboardingPreferences = {
+    USE_CASE: 'use_case',
```
Comment (AshishDhama) [inline utils/constants.jsx:136]: nit: Had discussion with @hmhealey would be better if we define constants near component 
https://community.mattermost.com/core/pl/qzjd3wuzf7yw9kygzjbcgz4yrr

**Comment by AshishDhama** [comment]
@neallred You have added a couple of SVG is it not possible to use them from https://mattermost.github.io/compass-icons/. Please ignore if those are logos. I think GitHub one we can use anyway.
@matthewbirtch Does it make sense to replace them with Icons
Screenshot 2022-02-14 at 12 45 10 AM
s

File: components/common/svg_images_components/boards_svg.tsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
```
[the remark this answers is not in the document]
Comment (neallred) [reply components/common/svg_images_components/boards_svg.tsx:1]: For a lot of them, its convention in the project more than anything. Some of them need different widths and heights, and we've chosen to style that as a prop on the component versus a classname. For a few of them (the channels preview svg is an example in this PR) there is dynamic content (the team and channel names).

[the remark this answers is not in the document]
Comment (JulienTant) [reply components/common/svg_images_components/boards_svg.tsx:1]: Thanks!

**Review by JulienTant — APPROVED** [review]
🧠💥 Good job! Thank you for your answers to my questions and the changes applied, LGTM.

**Review by esethna — APPROVED** [review]
Pair reviewed with Matt, nothing majorly blocking from my perspective aside from the UI polish items. Nice work @neallred!

**Comment by esethna** [comment]
@AshishDhama can you help re-review this to unblock merging?

**Comment by esethna** [comment]
@neallred are we okay to go ahead and merge this now? I think anything else we need can come in separate PRs unless there's a major blocking issue

**Comment by neallred** [comment]
Yep, we can merge now. Removed some last debugging code.

**Pull request merged**